src/scores/categorical/binary_impl.py

```diff
@@ -0,0 +1,149 @@
+"""
+This module contains methods for binary categories
+"""
+from collections.abc import Sequence
+from typing import Optional
+
+import numpy as np
+import xarray as xr
+
+from scores.functions import apply_weights
+from scores.processing import check_binary
+from scores.utils import gather_dimensions
+
+
+def probability_of_detection(
+    fcst: xr.DataArray,
+    obs: xr.DataArray,
+    reduce_dims: Optional[Sequence[str]] = None,
```

Comment:
I think preserve/reduce should be FlexibleDimensionTypes from scores.typing

Reply:
Updated and added test to test for xr.Datasets